### PR TITLE
performance-impl: start tracking lcp element

### PR DESCRIPTION
**Pull request by samouri**
**summary**

Creates a new csi beacon `lcpt` (Largest Contentful Paint Type) for tracking which type of element caused the LCP.

### CODE REVIEW

File: src/service/performance-impl.js [hunk 1]
```diff
@@ -839,6 +879,24 @@ export class Performance {
...
+ * Returns the given node if none are found.
+ *
+ * @param {!HTMLElement} node
+ * @return {!HTMLElement}
+ */
+function getOutermostAmpElement(node) {
+  let max = node;
+  while (node != null) {
+    node = node.parentNode;
+    if (node?.nodeName.startsWith('AMP-')) {
```
Comment (jridgewell) [inline src/service/performance-impl.js:893]: We've already asserted `!= null`.
```suggestion
    if (node.nodeName.startsWith('AMP-')) {
```

**Review by jridgewell — APPROVED** [review]

File: src/service/performance-impl.js (hunk 1 above)
Comment (samouri) [reply src/service/performance-impl.js]: Wait...no we didn't. At least not about the parent!

Comment (samouri) [reply src/service/performance-impl.js:893]: Reverting

File: src/service/performance-impl.js [hunk 2]
```diff
@@ -839,6 +879,24 @@ export class Performance {
...
+      max = node;
+    }
+  }
```
Comment (jridgewell) [inline src/service/performance-impl.js:896]:
```suggestion
  while ((node = node.parentNode) != null) {
    if (node.nodeName.startsWith('AMP-')) {
      max = node;
    }
  }
```

**Review by jridgewell — APPROVED** [review]

File: src/service/performance-impl.js (hunk 2 above)
Comment (samouri) [reply src/service/performance-impl.js:896]: or a do while

**Review by jridgewell — APPROVED** [review]

**Pull request merged**